ts/components/CompositionArea.tsx

```diff
@@ -357,6 +357,37 @@ export const CompositionArea = ({
     };
   }, [setLarge]);
 
+  // Listen to any key in the conversation panel to
+  React.useEffect(() => {
+    const handler = (e: KeyboardEvent) => {
+      const { key } = e;
+
+      // We don't want to react to a control character
+      if (key.length !== 1) {
+        return;
+      }
+
+      // We don't want to switch focus if another panel is up
+      const panels = document.querySelectorAll('.conversation .panel');
+      if (panels && panels.length > 1) {
+        return;
+      }
+
+      // We don't want to take focus away of input fields
+      const { activeElement } = document;
+      if (activeElement?.nodeName.toLowerCase() === 'input') {
+        return;
+      }
+
+      inputApiRef.current?.focus();
+    };
+
+    document.addEventListener('keydown', handler);
+    return () => {
+      document.removeEventListener('keydown', handler);
+    };
```

Comment:
Surfacing @josh-signal's comment here in a conversation:

> It's adding the event listener to `document` so yeah it's grabbing all input. Perhaps a ~better~ safer approach would be to figure out what node is focused and listen to `keypress` on that node? I think we can test the "in focus" issue using something like macOS' spaces.

Reply:
@josh-signal to be clear, you're suggesting this change?

```suggestion
    document.activeElement.addEventListener('keypress', handler);
    return () => {
      document.activeElement.removeEventListener('keypress', handler);
    };
```

Not sure if this will work, but at least it's a starting place for the conversation.